### PR TITLE
Make `HOMEBREW_TAP_PATH_REGEX` also match exact path.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

---

This way `Tap::from_path` also works for the exact path of a tap.

### CODE REVIEW

File: Library/Homebrew/tap_constants.rb [hunk 1]
```diff
@@ -5,6 +5,6 @@
 # match taps' directory paths, e.g. HOMEBREW_LIBRARY/Taps/someuser/sometap
 HOMEBREW_TAP_DIR_REGEX = %r{#{Regexp.escape(HOMEBREW_LIBRARY)}/Taps/(?<user>[\w-]+)/(?<repo>[\w-]+)}
 # match taps' formula paths, e.g. HOMEBREW_LIBRARY/Taps/someuser/sometap/someformula
-HOMEBREW_TAP_PATH_REGEX = Regexp.new(HOMEBREW_TAP_DIR_REGEX.source + %r{/(.*)}.source)
+HOMEBREW_TAP_PATH_REGEX = Regexp.new(HOMEBREW_TAP_DIR_REGEX.source + %r{(?:/(.*)|$)}.source)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/tap_constants.rb]: Can you provide an example string that doesn't match this? These two regexes look like they should be identical to me?

Comment (reitermarkus) [reply Library/Homebrew/tap_constants.rb]: The first one only matches sub-paths, the second one matches the exact path, without a trailing slash.

Comment (reitermarkus) [reply Library/Homebrew/tap_constants.rb]: It's not the same. `/?(.*)` is the same as just `(.*)`.

Comment (MikeMcQuaid) [reply Library/Homebrew/tap_constants.rb]: Sorry, I mean `/?(.*)$`

Comment (reitermarkus) [reply Library/Homebrew/tap_constants.rb]: That's still the same as `(.*)$`. The `?` should be around the whole thing:`(?:/(.*))?$`

Comment (MikeMcQuaid) [reply Library/Homebrew/tap_constants.rb]: Yeh, the greediness is different to what I thought 👍

**Pull request merged**